hyper/http20/stream.py

```diff
@@ -107,23 +108,26 @@ def send_data(self, data, final):
         sent, the ``final`` flag _must_ be set to True. If no data is to be
         sent, set ``data`` to ``None``.
         """
-        # Define a utility iterator for file objects.
-        def file_iterator(fobj):
+        if isinstance(data, types.GeneratorType):
+            for chunk in data:
+                for i in range(0, len(chunk), MAX_CHUNK):
+                    self._send_chunk(chunk[i:i + MAX_CHUNK], False)
+
+            # If ``final`` is True, send a empty chunk to end stream
+            if final:
+                self._send_chunk('', final)
+        elif hasattr(data, 'read'):
             while True:
-                data = fobj.read(MAX_CHUNK)
-                yield data
+                chunk = fobj.read(MAX_CHUNK)
                 if len(data) < MAX_CHUNK:
```

Comment:
You've changed the variable name here, so this is almost certainly not going to work.